modules/core/include/opencv2/core/cv_cpu_dispatch.h

```diff
@@ -152,6 +152,11 @@
 #  define CV_VSX3 1
 #endif
 
+#ifdef __mips_msa
+#  include "hal/msa_macros.h"
+#  define CV_MSA 1
```

Comment:
It is better to perform check via `CV_CPU_COMPILE_MSA` (at least, for debug purposes of SIMD optimizations this would allow to disable MSA via CMake)